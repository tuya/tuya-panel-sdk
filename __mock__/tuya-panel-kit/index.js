import React from 'react';
import { View } from 'react-native';
import TYText from './components/TYText';
import UnitText from './components/UnitText';

module.exports = {
  TYText,
  UnitText,
  Utils: {
    RatioUtils: {
      convertX: () => {},
    },
  },
};
