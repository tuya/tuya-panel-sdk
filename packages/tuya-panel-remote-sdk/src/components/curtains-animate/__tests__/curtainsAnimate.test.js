import React from 'react';
import { shallow } from 'enzyme';
import { Utils } from 'tuya-panel-kit';
import CurtainsAnimate from '../index';
import Res from './res';
const { roller, button, left, right } = Res;
const { convertX: cx } = Utils.RatioUtils;

describe('PressKey components', () => {
  it('basic render', () => {
    const wrapper = shallow(
      <CurtainsAnimate
        rollerImage={roller}
        buttonImage={button}
        curtainsLeftImage={left}
        curtainsRightImage={right}
        buttonPositionErrorValue={cx(4)}
        type={null}
      />
    );
    expect(wrapper).toMatchSnapshot();
    const wrapper1 = shallow(<CurtainsAnimate buttonPositionErrorValue={cx(4)} type={null} />);
    expect(wrapper1).toMatchSnapshot();
  });
  it('open render', () => {
    let type = null;
    const wrapper = shallow(
      <CurtainsAnimate
        rollerImage={roller}
        buttonImage={button}
        curtainsLeftImage={left}
        curtainsRightImage={right}
        buttonPositionErrorValue={cx(4)}
        type={type}
      />
    );
    const wrapper1 = shallow(<CurtainsAnimate buttonPositionErrorValue={cx(4)} type={type} />);
    type = 'open';
    wrapper.setProps({ type });
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
    wrapper1.setProps({ type });
    expect(wrapper1).toMatchSnapshot();
  });
  it('close render', () => {
    let type = null;
    const wrapper = shallow(
      <CurtainsAnimate
        rollerImage={roller}
        buttonImage={button}
        curtainsLeftImage={left}
        curtainsRightImage={right}
        buttonPositionErrorValue={cx(4)}
        type={type}
      />
    );
    const wrapper1 = shallow(<CurtainsAnimate buttonPositionErrorValue={cx(4)} type={type} />);
    type = 'close';
    wrapper.setProps({ type });
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
    wrapper1.setProps({ type });
    expect(wrapper1).toMatchSnapshot();
  });

  it('pause render', () => {
    let type = 'open';
    const wrapper = shallow(
      <CurtainsAnimate
        rollerImage={roller}
        buttonImage={button}
        curtainsLeftImage={left}
        curtainsRightImage={right}
        buttonPositionErrorValue={cx(4)}
        type={type}
      />
    );
    const wrapper1 = shallow(<CurtainsAnimate buttonPositionErrorValue={cx(4)} type={type} />);
    type = 'pause';
    wrapper.setProps({ type });
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
    wrapper1.setProps({ type });
    expect(wrapper1).toMatchSnapshot();
  });
});
