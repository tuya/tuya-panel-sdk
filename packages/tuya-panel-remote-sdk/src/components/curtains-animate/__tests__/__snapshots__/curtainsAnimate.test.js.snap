// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PressKey components basic render 1`] = `
<View
  style={
    Array [
      Object {
        "height": 560,
        "width": 560,
      },
      Object {
        "backgroundColor": "#fafafa",
      },
      null,
    ]
  }
>
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "height": 530,
          "left": 16,
          "top": 30,
          "width": 196,
        },
        null,
        null,
        null,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": "100%",
            "resizeMode": "stretch",
            "width": "100%",
          },
          null,
        ]
      }
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "height": 530,
          "right": 16,
          "top": 30,
          "width": 196,
        },
        null,
        null,
        null,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": "100%",
            "resizeMode": "stretch",
            "width": "100%",
          },
          null,
        ]
      }
    />
  </AnimatedComponent>
  <View
    style={
      Array [
        Object {
          "position": "absolute",
          "top": 16,
          "width": "100%",
        },
        null,
        null,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "resizeMode": "contain",
            "width": "100%",
          },
          null,
        ]
      }
    />
  </View>
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        null,
        Object {
          "borderRadius": 40,
          "height": 80,
          "left": 156,
          "width": 80,
        },
        null,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": "100%",
            "resizeMode": "contain",
            "width": "100%",
          },
          null,
        ]
      }
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        null,
        Object {
          "borderRadius": 40,
          "height": 80,
          "right": 156,
          "width": 80,
        },
        null,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": "100%",
            "resizeMode": "contain",
            "width": "100%",
          },
          null,
        ]
      }
    />
  </AnimatedComponent>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 80,
        "width": 560,
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderColor": "transparent",
            "height": 40,
          },
          Object {
            "height": 80,
            "width": 280,
          },
        ]
      }
    />
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderColor": "transparent",
            "height": 40,
          },
          Object {
            "height": 80,
            "width": 280,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`PressKey components basic render 2`] = `
<View
  style={
    Array [
      Object {
        "height": 560,
        "width": 560,
      },
      Object {
        "backgroundColor": "#fafafa",
      },
      null,
    ]
  }
>
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "height": 530,
          "left": 16,
          "top": 30,
          "width": 196,
        },
        Object {
          "borderWidth": 1,
        },
        null,
        null,
      ]
    }
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "height": 530,
          "right": 16,
          "top": 30,
          "width": 196,
        },
        null,
        Object {
          "borderWidth": 1,
        },
        null,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "position": "absolute",
          "top": 16,
          "width": "100%",
        },
        Object {
          "borderWidth": 1,
          "height": 40,
        },
        null,
      ]
    }
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "borderWidth": 1,
        },
        Object {
          "borderRadius": 40,
          "height": 80,
          "left": 156,
          "width": 80,
        },
        null,
      ]
    }
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "borderWidth": 1,
        },
        Object {
          "borderRadius": 40,
          "height": 80,
          "right": 156,
          "width": 80,
        },
        null,
      ]
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 80,
        "width": 560,
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderColor": "transparent",
            "height": 40,
          },
          Object {
            "height": 80,
            "width": 280,
          },
        ]
      }
    />
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderColor": "transparent",
            "height": 40,
          },
          Object {
            "height": 80,
            "width": 280,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`PressKey components close render 1`] = `
<View
  style={
    Array [
      Object {
        "height": 560,
        "width": 560,
      },
      Object {
        "backgroundColor": "#fafafa",
      },
      null,
    ]
  }
>
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "height": 530,
          "left": 16,
          "top": 30,
          "width": 196,
        },
        null,
        null,
        null,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": "100%",
            "resizeMode": "stretch",
            "width": "100%",
          },
          null,
        ]
      }
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "height": 530,
          "right": 16,
          "top": 30,
          "width": 196,
        },
        null,
        null,
        null,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": "100%",
            "resizeMode": "stretch",
            "width": "100%",
          },
          null,
        ]
      }
    />
  </AnimatedComponent>
  <View
    style={
      Array [
        Object {
          "position": "absolute",
          "top": 16,
          "width": "100%",
        },
        null,
        null,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "resizeMode": "contain",
            "width": "100%",
          },
          null,
        ]
      }
    />
  </View>
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        null,
        Object {
          "borderRadius": 40,
          "height": 80,
          "left": 156,
          "width": 80,
        },
        null,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": "100%",
            "resizeMode": "contain",
            "width": "100%",
          },
          null,
        ]
      }
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        null,
        Object {
          "borderRadius": 40,
          "height": 80,
          "right": 156,
          "width": 80,
        },
        null,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": "100%",
            "resizeMode": "contain",
            "width": "100%",
          },
          null,
        ]
      }
    />
  </AnimatedComponent>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 80,
        "width": 560,
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderColor": "transparent",
            "height": 40,
          },
          Object {
            "height": 80,
            "width": 280,
          },
        ]
      }
    />
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderColor": "transparent",
            "height": 40,
          },
          Object {
            "height": 80,
            "width": 280,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`PressKey components close render 2`] = `
<View
  style={
    Array [
      Object {
        "height": 560,
        "width": 560,
      },
      Object {
        "backgroundColor": "#fafafa",
      },
      null,
    ]
  }
>
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "height": 530,
          "left": 16,
          "top": 30,
          "width": 196,
        },
        Object {
          "borderWidth": 1,
        },
        null,
        null,
      ]
    }
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "height": 530,
          "right": 16,
          "top": 30,
          "width": 196,
        },
        null,
        Object {
          "borderWidth": 1,
        },
        null,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "position": "absolute",
          "top": 16,
          "width": "100%",
        },
        Object {
          "borderWidth": 1,
          "height": 40,
        },
        null,
      ]
    }
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "borderWidth": 1,
        },
        Object {
          "borderRadius": 40,
          "height": 80,
          "left": 156,
          "width": 80,
        },
        null,
      ]
    }
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "borderWidth": 1,
        },
        Object {
          "borderRadius": 40,
          "height": 80,
          "right": 156,
          "width": 80,
        },
        null,
      ]
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 80,
        "width": 560,
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderColor": "transparent",
            "height": 40,
          },
          Object {
            "height": 80,
            "width": 280,
          },
        ]
      }
    />
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderColor": "transparent",
            "height": 40,
          },
          Object {
            "height": 80,
            "width": 280,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`PressKey components open render 1`] = `
<View
  style={
    Array [
      Object {
        "height": 560,
        "width": 560,
      },
      Object {
        "backgroundColor": "#fafafa",
      },
      null,
    ]
  }
>
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "height": 530,
          "left": 16,
          "top": 30,
          "width": 196,
        },
        null,
        null,
        null,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": "100%",
            "resizeMode": "stretch",
            "width": "100%",
          },
          null,
        ]
      }
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "height": 530,
          "right": 16,
          "top": 30,
          "width": 196,
        },
        null,
        null,
        null,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": "100%",
            "resizeMode": "stretch",
            "width": "100%",
          },
          null,
        ]
      }
    />
  </AnimatedComponent>
  <View
    style={
      Array [
        Object {
          "position": "absolute",
          "top": 16,
          "width": "100%",
        },
        null,
        null,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "resizeMode": "contain",
            "width": "100%",
          },
          null,
        ]
      }
    />
  </View>
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        null,
        Object {
          "borderRadius": 40,
          "height": 80,
          "left": 156,
          "width": 80,
        },
        null,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": "100%",
            "resizeMode": "contain",
            "width": "100%",
          },
          null,
        ]
      }
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        null,
        Object {
          "borderRadius": 40,
          "height": 80,
          "right": 156,
          "width": 80,
        },
        null,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": "100%",
            "resizeMode": "contain",
            "width": "100%",
          },
          null,
        ]
      }
    />
  </AnimatedComponent>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 80,
        "width": 560,
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderColor": "transparent",
            "height": 40,
          },
          Object {
            "height": 80,
            "width": 280,
          },
        ]
      }
    />
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderColor": "transparent",
            "height": 40,
          },
          Object {
            "height": 80,
            "width": 280,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`PressKey components open render 2`] = `
<View
  style={
    Array [
      Object {
        "height": 560,
        "width": 560,
      },
      Object {
        "backgroundColor": "#fafafa",
      },
      null,
    ]
  }
>
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "height": 530,
          "left": 16,
          "top": 30,
          "width": 196,
        },
        Object {
          "borderWidth": 1,
        },
        null,
        null,
      ]
    }
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "height": 530,
          "right": 16,
          "top": 30,
          "width": 196,
        },
        null,
        Object {
          "borderWidth": 1,
        },
        null,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "position": "absolute",
          "top": 16,
          "width": "100%",
        },
        Object {
          "borderWidth": 1,
          "height": 40,
        },
        null,
      ]
    }
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "borderWidth": 1,
        },
        Object {
          "borderRadius": 40,
          "height": 80,
          "left": 156,
          "width": 80,
        },
        null,
      ]
    }
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "borderWidth": 1,
        },
        Object {
          "borderRadius": 40,
          "height": 80,
          "right": 156,
          "width": 80,
        },
        null,
      ]
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 80,
        "width": 560,
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderColor": "transparent",
            "height": 40,
          },
          Object {
            "height": 80,
            "width": 280,
          },
        ]
      }
    />
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderColor": "transparent",
            "height": 40,
          },
          Object {
            "height": 80,
            "width": 280,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`PressKey components pause render 1`] = `
<View
  style={
    Array [
      Object {
        "height": 560,
        "width": 560,
      },
      Object {
        "backgroundColor": "#fafafa",
      },
      null,
    ]
  }
>
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "height": 530,
          "left": 16,
          "top": 30,
          "width": 196,
        },
        null,
        null,
        null,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": "100%",
            "resizeMode": "stretch",
            "width": "100%",
          },
          null,
        ]
      }
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "height": 530,
          "right": 16,
          "top": 30,
          "width": 196,
        },
        null,
        null,
        null,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": "100%",
            "resizeMode": "stretch",
            "width": "100%",
          },
          null,
        ]
      }
    />
  </AnimatedComponent>
  <View
    style={
      Array [
        Object {
          "position": "absolute",
          "top": 16,
          "width": "100%",
        },
        null,
        null,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "resizeMode": "contain",
            "width": "100%",
          },
          null,
        ]
      }
    />
  </View>
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        null,
        Object {
          "borderRadius": 40,
          "height": 80,
          "left": 156,
          "width": 80,
        },
        null,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": "100%",
            "resizeMode": "contain",
            "width": "100%",
          },
          null,
        ]
      }
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        null,
        Object {
          "borderRadius": 40,
          "height": 80,
          "right": 156,
          "width": 80,
        },
        null,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": "100%",
            "resizeMode": "contain",
            "width": "100%",
          },
          null,
        ]
      }
    />
  </AnimatedComponent>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 80,
        "width": 560,
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderColor": "transparent",
            "height": 40,
          },
          Object {
            "height": 80,
            "width": 280,
          },
        ]
      }
    />
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderColor": "transparent",
            "height": 40,
          },
          Object {
            "height": 80,
            "width": 280,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`PressKey components pause render 2`] = `
<View
  style={
    Array [
      Object {
        "height": 560,
        "width": 560,
      },
      Object {
        "backgroundColor": "#fafafa",
      },
      null,
    ]
  }
>
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "height": 530,
          "left": 16,
          "top": 30,
          "width": 196,
        },
        Object {
          "borderWidth": 1,
        },
        null,
        null,
      ]
    }
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "height": 530,
          "right": 16,
          "top": 30,
          "width": 196,
        },
        null,
        Object {
          "borderWidth": 1,
        },
        null,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "position": "absolute",
          "top": 16,
          "width": "100%",
        },
        Object {
          "borderWidth": 1,
          "height": 40,
        },
        null,
      ]
    }
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "borderWidth": 1,
        },
        Object {
          "borderRadius": 40,
          "height": 80,
          "left": 156,
          "width": 80,
        },
        null,
      ]
    }
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "borderWidth": 1,
        },
        Object {
          "borderRadius": 40,
          "height": 80,
          "right": 156,
          "width": 80,
        },
        null,
      ]
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 80,
        "width": 560,
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderColor": "transparent",
            "height": 40,
          },
          Object {
            "height": 80,
            "width": 280,
          },
        ]
      }
    />
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderColor": "transparent",
            "height": 40,
          },
          Object {
            "height": 80,
            "width": 280,
          },
        ]
      }
    />
  </View>
</View>
`;
